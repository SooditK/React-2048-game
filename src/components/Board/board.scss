@import '../../assets/styles/vars';

.board {
  border-collapse: separate;
  border-spacing: 0.25vw;
}

@media all and (max-width: $break-point) {
  .board {
    border-spacing: 1vw;
  }
}
