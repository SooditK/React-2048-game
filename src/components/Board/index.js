import React from 'react';
import PropTypes from 'prop-types';
import Row from '../Row';
import styles from './board.scss';
import { isObjEqual } from '../../utils/helpers';

// Game board
export default class Board extends React.Component {
  static propTypes = {
    matrix: PropTypes.arrayOf(PropTypes.array).isRequired
  };

  shouldComponentUpdate(nextProps, nextState) {
    return (
      !isObjEqual(nextProps, this.props) || !isObjEqual(nextState, this.state)
    );
  }

  render() {
    const {
      props: { matrix }
    } = this;

    return (
      <table className={styles.board}>
        <tbody>
          {/* eslint-disable react/no-array-index-key */}
          {matrix.map((row, idx) => (
            <Row row={row} key={idx} />
          ))}
        </tbody>
      </table>
    );
  }
}
