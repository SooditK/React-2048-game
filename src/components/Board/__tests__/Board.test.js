import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Board from '..';

Enzyme.configure({ adapter: new Adapter() });

const MATRIX = [[2, 0, 0, 0], [0, 0, 0, 0], [0, 0, 4, 0], [0, 0, 0, 0]];

describe('<Board />', () => {
  it('component render', () => {
    const board = renderer.create(<Board matrix={MATRIX} />).toJSON();
    expect(board).toMatchSnapshot();
  });

  it('shouldComponentUpdate', () => {
    const speaker = mount(<Board matrix={MATRIX} />);
    expect(speaker.props().matrix).toEqual(MATRIX);
    let shouldUpdate = speaker.instance().shouldComponentUpdate(
      {
        matrix: MATRIX
      },
      null
    );
    expect(shouldUpdate).toBe(false);
    shouldUpdate = speaker.instance().shouldComponentUpdate(
      {
        matrix: [[2, 2, 2, 2], [0, 0, 0, 0], [0, 0, 4, 0], [0, 0, 0, 0]]
      },
      null
    );
    expect(shouldUpdate).toBe(true);
  });
});
