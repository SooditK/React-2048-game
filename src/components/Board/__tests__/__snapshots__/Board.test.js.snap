// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Board /> component render 1`] = `
<table
  className="board"
>
  <tbody>
    <tr>
      <td>
        <div
          className="cell color-2"
        >
          <div
            className="number"
          >
            2
          </div>
        </div>
      </td>
      <td>
        <div
          className="cell"
        >
          <div
            className="number"
          />
        </div>
      </td>
      <td>
        <div
          className="cell"
        >
          <div
            className="number"
          />
        </div>
      </td>
      <td>
        <div
          className="cell"
        >
          <div
            className="number"
          />
        </div>
      </td>
    </tr>
    <tr>
      <td>
        <div
          className="cell"
        >
          <div
            className="number"
          />
        </div>
      </td>
      <td>
        <div
          className="cell"
        >
          <div
            className="number"
          />
        </div>
      </td>
      <td>
        <div
          className="cell"
        >
          <div
            className="number"
          />
        </div>
      </td>
      <td>
        <div
          className="cell"
        >
          <div
            className="number"
          />
        </div>
      </td>
    </tr>
    <tr>
      <td>
        <div
          className="cell"
        >
          <div
            className="number"
          />
        </div>
      </td>
      <td>
        <div
          className="cell"
        >
          <div
            className="number"
          />
        </div>
      </td>
      <td>
        <div
          className="cell color-4"
        >
          <div
            className="number"
          >
            4
          </div>
        </div>
      </td>
      <td>
        <div
          className="cell"
        >
          <div
            className="number"
          />
        </div>
      </td>
    </tr>
    <tr>
      <td>
        <div
          className="cell"
        >
          <div
            className="number"
          />
        </div>
      </td>
      <td>
        <div
          className="cell"
        >
          <div
            className="number"
          />
        </div>
      </td>
      <td>
        <div
          className="cell"
        >
          <div
            className="number"
          />
        </div>
      </td>
      <td>
        <div
          className="cell"
        >
          <div
            className="number"
          />
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;
